exasol_advanced_analytics_framework/lua/src/event_loop.lua

```diff
@@ -0,0 +1,58 @@
+---
+-- @module event_loop
+--
+-- This module processes only the state transitions by executing queries returned by the Event Handler
+--
+
+
+
+local M = {}
+local exa_error = require("exaerror")
+
+_G.global_env = {
+    pquery = pquery,
+    error = error
+}
+
+---
+-- Executes the given set of queries.
+--
+function M._run_queries(queries, from_index)
+    local all_success = true
+    for i=from_index, #queries do
+        local success, result = _G.global_env.pquery(queries[i][1])
+        all_success = all_success and success
+        if not success then
+            local error_obj = exa_error.create(
+                    "E-AAF-3",
+                    "Error occurred in executing queries: " .. result.error_message)
+            _G.global_env.error(tostring(error_obj))
```

Comment:
if you mean "error handling", I also thought it. But at this stage of the development it might be earlier to handle it. We can do it when the implementation is matured.